### PR TITLE
=str: various minor cleanups

### PR DESCRIPTION
**Pull request by aruediger**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala [hunk 1]
```diff
@@ -174,7 +174,7 @@ trait ActorPublisher[T] extends Actor {
    * otherwise `onNext` will throw `IllegalStateException`.
    */
   def onNext(element: T): Unit = lifecycleState match {
-    case Active | PreSubscriber ⇒
+    case Active | PreSubscriber | CompleteThenStop ⇒
```
Comment (2beaucoup) [inline akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala]: Not sure if this is the right place for `CompleteThenStop ` (3x).

File: akka-stream/src/main/scala/akka/stream/impl/io/SslTlsCipherActor.scala [hunk 2]
```diff
@@ -156,12 +149,10 @@ private[akka] class SslTlsCipherActor(settings: ActorMaterializerSettings, sslCo
     e
   }
   var currentSession = engine.getSession
-  var currentSessionParameters = firstSession
```
Comment (2beaucoup) [inline akka-stream/src/main/scala/akka/stream/impl/io/SslTlsCipherActor.scala]: switched from `var` to function arg in `applySessionParameters`.

**Comment by drewhk** [comment]
OK TO TEST

**Comment by drewhk** [comment]
and LGTM

**Comment by ktoso** [comment]
This needs a rebase sadly.

**Comment by 2beaucoup** [comment]
ok, done.

**Pull request merged**